src/Core/Localization/Currency/Repository.php

```diff
@@ -114,7 +114,7 @@ private function createCurrencyFromData(CurrencyData $currencyData)
             $currencyData->isActive(),
             $currencyData->getConversionRate(),
             $currencyData->getIsoCode(),
-            $currencyData->getNumericIsoCode(),
+            (int) $currencyData->getNumericIsoCode(),
```

Comment:
I think we already had this conversation but `numericIsoCode` can be null for unofficial currencies So even if in this context it seems the data comes from CLDR references and is not likely to be null forcing it to an int can lead to unexpected behaviour in the future

Especially since CurrencyData relies on the null value in its override mechanism 

If something needs to be changes it might be in `PrestaShop\PrestaShop\Core\Localization\Currency` that numeric is code is `string|null`